cluster-essentials/pkg/client/helpers.go

```diff
@@ -0,0 +1,264 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	yttui "github.com/vmware-tanzu/carvel-ytt/pkg/cmd/ui"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/files"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/workspace"
+	"github.com/vmware-tanzu/carvel-ytt/pkg/workspace/datavalues"
+
+	"github.com/cppforlife/go-cli-ui/ui"
+	"github.com/k14s/kbld/pkg/kbld/cmd"
+
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/bundle"
+	imgpkgcmd "github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/cmd"
+	"github.com/vmware-tanzu/carvel-imgpkg/pkg/imgpkg/registry"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+const ConfigFilePermissions = 0o600
+
+func DownloadBundle(bundlePath, outputDir string) error {
```

Comment:
Is there a reason all the functions in this file are exported? They all look like something you just use in `Install`. Please un-export if there is no usage outside this package.